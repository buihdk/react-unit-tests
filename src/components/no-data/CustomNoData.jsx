import React from 'react';
import PropTypes from 'prop-types';

const styles = {};

const CustomNoData = props => {
  const { noDataText, createNewDataText, noDataLogo } = props;

  return (
    <div className={styles.container}>
      <div className={styles.noDataLogo}> {noDataLogo} </div>
      <div className={styles.textStyle}>{noDataText}</div>
      <div className={styles.newDataTextStyle}> {createNewDataText} </div>
    </div>
  );
};

CustomNoData.propTypes = {
  noDataText: PropTypes.string,
  createNewDataText: PropTypes.string,
  noDataLogo: PropTypes.node,
};

CustomNoData.defaultProps = {
  noDataText: '',
  createNewDataText: '',
  noDataLogo: null,
};

export default CustomNoData;
