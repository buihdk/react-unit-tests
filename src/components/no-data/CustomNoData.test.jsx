import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CustomNoData from './CustomNoData';

describe('CustomNoData', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(<CustomNoData />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
