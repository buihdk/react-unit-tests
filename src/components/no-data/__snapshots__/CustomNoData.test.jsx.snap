// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomNoData renders without crashing 1`] = `
<div>
  <div>
     
     
  </div>
  <div />
  <div>
     
     
  </div>
</div>
`;
